zap/src/config.rs

```diff
@@ -291,6 +292,27 @@ impl<'src> Ty<'src> {
 			Self::Color3 => (12, Some(12)),
 			Self::Vector2 => (8, Some(8)),
 			Self::Vector3 => (12, Some(12)),
+			Self::Vector(x_ty, y_ty, z_ty) => {
+				let x_size = match **x_ty {
+					Ty::Num(numty, _) => numty.size(),
+					_ => unreachable!(),
```

Comment:
After testing locally, this `unreachable`/`panic` *does* trigger before the diagnostic is reported.
Maybe we should instead do:
```suggestion
					_ => 0,
```